main.py

```diff
@@ -2,12 +2,25 @@
 from prompt_toolkit import print_formatted_text, HTML
 from websockets.exceptions import ConnectionClosedError
 from utils.helper_display import HelperDisplay
+from cryptography.fernet import Fernet, InvalidToken
 
 
 USERS = {}
+CHATROOM = {}
 sepr = chr(969696)
+current_rooms=[]
 helper_display = HelperDisplay()
 
+class fernetst():
+    def __init__(self, pswd):
+        self.suit = Fernet(pswd)
+
+    def encrjson(self, data):
+        return self.suit.encrypt(data.encode("utf8")).decode("utf8")
+
+    def decrjson(self, data):
+        return self.suit.decrypt(data.encode("utf8")).decode("utf8")
```

Comment:
This totally defeats the purpose of end-to-end encryption. The server is able to read and understand the messages conveyed among the users.

Reply:
Well, I wasn't sure of that part either. But without generating the credentials at the server side, we always have risk of inconsistency and doing this at server side also prevents from creating same named chatroom. Also, in the earlier method of checking credentials we were justing checking if the password are valid ,not if they are correct. There may happen that a person enters a chatroom 12345678 with a wrong password(unknowingly) and then send some text(which he thinks is received at the other end) but in reality the other side is just reading some gibberish. I wasn't sure how to handle this. Can I have a discussion with others as well on this. Please correct if I am wrong.